Source/CombatExtended/CombatExtended/Projectiles/ProjectileCE_Airburst.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Verse;
+using RimWorld;
+using CombatExtended.Utilities;
+
+namespace CombatExtended
+{
+    public class ProjectileCE_Airburst : ProjectileCE
+    {
+        public int updateRate => props.updateRate;
+
+        public int timer = -10;
+        
+        public bool detonated;
+
+        public ModExtension_Airburst props => this.def.GetModExtension<ModExtension_Airburst>();
+        public override void Tick()
+        {
+            if (timer >= updateRate)
+            {
+                 if (!detonated)
+                {
+                    if (timer >= updateRate)
+                    {
+                        UpdateDistance();
+                        timer = 0;
+                    }
+                    timer++;
+                    base.Tick();
+                }
+            }
+        }
+
+        public virtual void UpdateDistance()
+        {
+             if (
+                detonated || 
+                this.Position.DistanceTo(this.intendedTarget.Cell) <= props.distToExplode
+                ||
+                this.Position.PawnsInRange(this.Map, props.distToExplode).Any()

```

Comment:
Are there circumstances were this will cause the projectile to explode on launch?  Looks like there's a minimum number of ticks before it "arms", but low velocity explosives might be limited with this.  It *could* use the pawns-along-line code used by raycast suppression to only look for pawns in front of it.